tests/ttnn/unit_tests/operations/test_moreh_adam.py

```diff
@@ -141,3 +144,24 @@ def forward(self, x):
         logger.debug(f"Out passing (max_exp_avg_sq)={passing}")
         logger.debug(f"Output pcc={out}")
     assert passing
+
+
+@pytest.mark.parametrize(
+    "params",
+    (
+        # shape, lr, betas, eps, weight_decay, amsgrad, fp32_dest_acc_en
+        ([32, 32], 0.0, (0.9, 0.999), 1e-06, 0.0, True, True),
+        ([2, 2, 2, 2, 2, 2, 64, 64], 0.0, (0.9, 0.999), 1e-06, 0.0, False, False),
+    ),
+)
+def test_moreh_adam_enable_cache(params, device, use_program_cache):
+    for i in range(4):
+        if i % 2 == 1:
+            param_list = list(params)
+            param_list[5] = False if param_list[5] else True
+            params = tuple(param_list)
+
+        shape, lr, betas, eps, weight_decay, amsgrad, fp32_dest_acc_en = params
```

Comment:
I have reviewed the changes, and since the output buffer changes based on whether amsgrad is set to True or False, it makes sense to test both cases. This approach looks good.

 However, what is the reason that only amsgrad needs to be changed inside the for loop?
```suggestion
        shape, lr, betas, eps, weight_decay, amsgrad, fp32_dest_acc_en = params
        if i % 2 == 1:
            amsgrad = False if amsgrad else True
```
- If it needs to be changed, it would be better to use the variable name amsgrad rather than param_list[5].

Reply:
I've made the changes required